main.tf

```diff
@@ -8,6 +8,7 @@ moved {
 }
 
 resource "tls_private_key" "ssh" {
+  count = var.admin_username == null ? 0 : 1
```

Comment:
Since we have added `count` to this resource we need add index to the reference to `tls_provate.key.ssh`.

Reply:
You are right. I added the index reference. I will have to test if this block will work after this change:
```
moved {
  from = module.ssh-key.tls_private_key.ssh
  to   = tls_private_key.ssh
}
```

Would be a valid

Reply:
I think it's a tricky scenario since the original code declare `tls_private_key` resource unconditionally, what if a user want to migrate the existing state to new version meanwhile `var.admin_username` is `null`? We might need an upgrade test to see what'll happen in this circumstance.